src/app/[lng]/learn/[slug]/page.tsx

```diff
@@ -1,8 +1,50 @@
 import { getArticleBySlug } from '@/app/lib/getArticleBySlug';
 import { getArticlesByTag } from '@/app/lib/getArticlesByTag';
 import LearnArticlePage from '@/app/ui/learn/LearnArticlePage';
-import type { BlogArticleData } from '@/types/strapi';
+import type { BlogArticleAttributes, BlogArticleData } from '@/types/strapi';
 import { getCookies } from '@/app/lib/getCookies';
+import { Metadata } from 'next';
+import { sliceStrToXChar } from 'src/utils/splitStringToXChar';
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { slug: string };
+}): Promise<Metadata> {
+  try {
+    const article = await getArticleBySlug(params.slug);
+
+    if (!article.data || !article.data.data?.[0]) {
+      throw new Error();
+    }
+
+    const articleData = article.data.data?.[0]
+      .attributes as BlogArticleAttributes;
+
+    return {
+      title: `Jumper Learn | ${sliceStrToXChar(articleData.Title, 45)}`,
+      description: articleData.Subtitle,
+      openGraph: {
+        title: `Jumper Learn | ${sliceStrToXChar(articleData.Title, 45)}`,
+        description: `${sliceStrToXChar(articleData.Subtitle, 60)}`,
+        images: [
+          {
+            url: `${article.url}${articleData.Image.data.attributes?.url}`,
+            width: 900,
+            height: 450,
+            alt: 'banner image',
+          },
+        ],
+        type: 'article',
+      },
+    };
+  } catch (err) {
+    return {
+      title: `Jumper Learn | ${sliceStrToXChar(params.slug.replaceAll('-', ' '), 45)}`,
```

Comment:
Usually, the slug and title seem to be similar for the most part. The advantage of using the title seems to be that it also comes with uppercase letters on some strings which may look little more polished. Also, mostly we don´t use those dashes in titles, so there might not be a need for the replaceAll. 
Wdyt?